common/rpc/factory.go

```diff
@@ -49,19 +48,14 @@ type RPCFactory struct {
 	dispatcher *yarpc.Dispatcher
 }
 
-// NewFactory builds a new RPCFactory
+// NewFactory builds a new rpc.Factory
 // conforming to the underlying configuration
-func (cfg *RPC) NewFactory(sName string, logger log.Logger, grpcPorts GRPCPorts) *RPCFactory {
-	return newRPCFactory(cfg, sName, logger, grpcPorts)
-}
-
-func newRPCFactory(cfg *RPC, sName string, logger log.Logger, grpcPorts GRPCPorts) *RPCFactory {
-	factory := &RPCFactory{config: cfg, serviceName: sName, logger: logger, grpcPorts: grpcPorts}
-	return factory
+func NewFactory(sName string, cfg config.RPC, logger log.Logger, grpcPorts GRPCPorts) *Factory {
```

Comment:
nit and not directly related to this change: sName -> name or service

Reply:
Agree, fixed.